ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -444,6 +463,26 @@ public String toString() {
           + " , nodes: " + nodes
           + " } ";
     }
+
+    @Override
+    public boolean equals(final Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      final PartitionLocation that = (PartitionLocation) o;
+      return partition == that.partition
+          && Objects.equals(keys, that.keys)
+          && Objects.equals(nodes, that.nodes);
```

Comment:
Since nodes are stored in a list, do we guarantee the ordering is always the same (otherwise they would be treated as not equal)?

Reply:
good catch, we do not guarantee ordering - this was used only for testing so it wasn't caught. I'll change it to a set comparison